paddle/fluid/operators/momentum_op.cc

```diff
@@ -110,6 +127,8 @@ else:   \\
 }  // namespace paddle
 
 namespace ops = paddle::operators;
-REGISTER_OP_WITHOUT_GRADIENT(momentum, ops::MomentumOp, ops::MomentumOpMaker);
+REGISTER_OPERATOR(momentum, ops::MomentumOp, ops::MomentumOpMaker,
```

Comment:
momentum_op is optimizer, has no gradient operator.

Reply:
I agree, so I use EmptyGradOpMaker. Switch from REGISTER_OP_WITHOUT_GRADIENT to REGISTER_OPERATOR is to support OpInferVarType.